client/service/src/builder.rs

```diff
@@ -487,8 +491,66 @@ where
 	)
 	.map_err(|e| Error::Application(Box::new(e)))?;
 
+	let database_path = match config.database {
+		sc_client_db::DatabaseSource::Auto { ref paritydb_path, ref rocksdb_path, .. } =>
+			if rocksdb_path.exists() {
+				Some(rocksdb_path.as_path())
+			} else {
+				Some(paritydb_path.as_path())
+			},
+		sc_client_db::DatabaseSource::RocksDb { ref path, .. } |
+		sc_client_db::DatabaseSource::ParityDb { ref path, .. } => Some(path.as_path()),
+		sc_client_db::DatabaseSource::Custom { .. } => None,
+	};
+
+	let sysinfo = crate::sysinfo::gather_sysinfo();
+	let hwbench = crate::sysinfo::gather_hwbench(database_path);
+
+	info!("💻 Operating system: {}", TARGET_OS);
+	info!("💻 CPU architecture: {}", TARGET_ARCH);
+	if !TARGET_ENV.is_empty() {
+		info!("💻 Target environment: {}", TARGET_ENV);
+	}
```

Comment:
Just looking at these log lines, I would get the impression that the properties of the running system are listed, not those of the build target. I know it is really an edge-case, but foreign ELF formats can be loaded and run emulated on a different system. Are we okay with ignoring those fringe usages?

Reply:
Hmm... well, that is a good point; I don't think we have to care about this *in general* though since those should mostly be *really* fringe cases, and detecting this will most likely not be easy. (That said, if anyone has any counterpoints here or any good ideas how to handle this in a reasonable way I'm all ears.)

I guess the most likely cases here would be either someone running the Linux binary on a BSD system, or someone running an amd64 binary on an M1 Mac (but we don't provide binaries for macOS, so they'd have to compile it themselves, and if they're compiling it themselves then why not compile a native aarch64 binary in the first place and run that?).